examples/src/bin/ech-client.rs

```diff
@@ -43,18 +43,28 @@ use rustls::RootCertStore;
 async fn main() {
     let args = Args::parse();
 
-    // Find raw ECH configs using DNS-over-HTTPS with Hickory DNS.
-    let resolver_config = if args.use_cloudflare_dns {
-        ResolverConfig::cloudflare_https()
-    } else {
-        ResolverConfig::google_https()
-    };
-    let resolver = Resolver::tokio(resolver_config, ResolverOpts::default());
-    let server_ech_config = match args.grease {
-        true => None, // Force the use of the GREASE ext by skipping ECH config lookup
+    // Determine the ECH config to use.
```

Comment:
This commit feels like it has a bunch of stuff going on at the same time... Maybe split it a little?

Reply:
Broken up into smaller pieces now. I should have done that from the start :-)